packages/unlock-js/src/walletService.ts

```diff
@@ -1171,6 +1173,40 @@ export default class WalletService extends UnlockService {
     if (!params.tokenId) throw new Error('Missing tokenId')
 
     const version = await this.lockContractAbiVersion(params.lockAddress)
+
+    if (params.time) {
+      if (!version.shareKey) {
+        throw new Error('Lock version not supported')
+      }
```

Comment:
ok so the way to implement this is to *not* do that in `walletService.ts` but to add a `shareKey.ts` and `lendKey.ts` file to implement this based on the versions (check how other methods are working!)